src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterEnum.cs

```diff
@@ -10,6 +10,15 @@ internal sealed class JsonValueConverterEnum<TValue> : JsonValueConverter<TValue
         where TValue : struct, Enum
     {
         private static readonly bool s_isUint64 = Enum.GetUnderlyingType(typeof(TValue)) == typeof(ulong);
+        private static readonly bool s_isInt64 = Enum.GetUnderlyingType(typeof(TValue)) == typeof(long);
```

Comment:
Maybe this can instead be a switch on the underlying type (and we can remove the statics). Is this an expensive call compared to the other work the enum converter has to do?

Reply:
I thought that had been to take advantage jitter optimization...I mean it should statically recognize type and elide branches.
@jkotas do you know if jitter can do this?

Reply:
The JIT can do this for simple comparisons of two `typeof`, etc. This is not one of those - it calls `Enum.GetUnderlyingType` that the JIT optimizations know nothing about.

Caching these in `readonly` statics is fine.